File: app/models/cmd/webhook.go
package cmd

import (
	"github.com/Spicy-Bush/fider-tarkov-community/app/models/dto"
	"github.com/Spicy-Bush/fider-tarkov-community/app/models/enum"
	"github.com/Spicy-Bush/fider-tarkov-community/app/pkg/webhook"
)

type TestWebhook struct {
	ID int

	Result *dto.WebhookTriggerResult
}

type TriggerWebhooks struct {
	Type  enum.WebhookType
	Props webhook.Props
}

type PreviewWebhook struct {
	Type    enum.WebhookType
	Url     string
	Content string

	Result *dto.WebhookPreviewResult
}

type GetWebhookProps struct {
	Type enum.WebhookType

	Result webhook.Props
}

File: public/services/actions/webhook.ts
import { http, Result, StringObject } from "@fider/services"
import { WebhookData, WebhookPreviewResult, WebhookTriggerResult, WebhookType } from "@fider/models"

export const createWebhook = async (data: WebhookData): Promise<Result<{ id: number }>> => {
  return await http.post(`/_api/admin/webhook`, data)
}

export const updateWebhook = async (id: number, data: WebhookData): Promise<Result> => {
  return await http.put(`/_api/admin/webhook/${id}`, data)
}

export const deleteWebhook = async (id: number): Promise<Result> => {
  return await http.delete(`/_api/admin/webhook/${id}`)
}

export const testWebhook = async (id: number): Promise<Result<WebhookTriggerResult>> => {
  return await http.get(`/_api/admin/webhook/test/${id}`)
}

export const previewWebhook = async (type: WebhookType, url: string, content: string): Promise<Result<WebhookPreviewResult>> => {
  return await http.post("/_api/admin/webhook/preview", { type, url, content })
}

export const getWebhookHelp = async (type: WebhookType): Promise<Result<StringObject>> => {
  return await http.get(`/_api/admin/webhook/props/${type}`)
}
